Form validation

wontoan opened the pull request:
-My first sign up form with basic validation!

wontoan committed:
Fixed /signup page throwing error.

wontoan committed:
Fixed /login page throwing error.

vishalarora committed:
Ignore config/database.yml

wontoan committed:
Merge branch 'add_user_login_forms' into form-validation

wontoan committed:
Merge branch 'styleguide-buttons' into form-validation

wontoan committed:
Merge branch 'styleguide' into form-validation

wontoan committed:
Re-design log in form using style guide & flexbox

wontoan committed:
Re-design signup form with style guide & flex box

wontoan committed:
Design for Login & Signup forms solidified

To Do: Replace Input fields with inputs from styleguide

wontoan committed:
Adjusted button-sizing, added submit class to create accnt/login buttons

vishalarora committed:
Merge branch 'styleguide-inputs' into form-validation

wontoan committed:
Form validation test

vishalarora committed:
Merge remote-tracking branch 'origin/form-validation' into form-validation

vishalarora committed:
Fire JS only after page has loaded

wontoan committed:
Email is now validating

wontoan committed:
switched <input> back to <button>

wontoan committed:
Full Name and Password are now validating as well (See description)

-Full name cannot contain special characters except (-) and (‘) (for
italian and hyphenated names).
-Password must be at least 8 characters in length and contain no
spaces.

wontoan committed:
Merged 'alerts' branch + added 'error' styling to validation.

wontoan committed:
Update deadlines

wontoan committed:
V.1 of Sign Up Form + Validation

wontoan committed:
Revert "Update deadlines"

This reverts commit 225940530910415a89213a9ee9570f998fca09dd.

wontoan committed:
Move alert stylesheet to components/alerts

vishalarora commented:
@wontoan Great work.  The validation works exactly as it should.  One small point though:  the point of doing client-side (JS) validation on top of the existing server-side validation (in Rails) is to allow for quicker feedback and save a request to the server.  

The way you've done is acceptable but it should either actually submit the form once the inputs are validated (instead of showing the "success" alert) or do the validation inline (i.e. as the user is typing).

vishalarora commented on app/assets/javascripts/components/signup.js:
```diff
@@ -0,0 +1,49 @@
+
+  $(".signupForm").submit(function(event) {
+
+  var errorMessage = "";
```
Your spacing/indenting is off here.

vishalarora commented on app/assets/javascripts/components/signup.js:
```diff
@@ -0,0 +1,49 @@
+  function isValidFullName(fullName) {
+    var validate_name = new RegExp (/^[a-zA-Z ]+$/);
+    return validate_name.test(fullName);
+  }
```
The `isValid...` functions can be scoped outside the signup form submit function.  You'll probably want to reuse some these on the login form.

vishalarora commented on app/assets/javascripts/components/signup.js:
```diff
@@ -0,0 +1,49 @@
+
+  function isValidEmailAddress(emailAddress) {
+
+    var pattern = new RegExp(/^((([a-z]|\d|[!#\$%&'\*\+\-\/=\?\^_`{\|}~]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+(\.([a-z]|\d|[!#\$%&'\*\+\-\/=\?\^_`{\|}~]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+)*)|((\x22)((((\x20|\x09)*(\x0d\x0a))?(\x20|\x09)+)?(([\x01-\x08\x0b\x0c\x0e-\x1f\x7f]|\x21|[\x23-\x5b]|[\x5d-\x7e]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(\\([\x01-\x09\x0b\x0c\x0d-\x7f]|[\u00 ...
```
This is the official standard for what an email should be but it only checks for basic syntax.  So an email address like `toan@wontons.are.awesome` would get past this filter even though it's obviously invalid.  Also, this RegEx allows `""` and `[]` which can break some email clients/scripts.

My preferred RegEx for email validation is this:

```
[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@
(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+(?:[A-Z]{2}|com|org|net|edu|gov|mil|
biz|info|mobi|name|aero|asia|jobs|museum)\b
```

This removes quotes and square brackets and limits the TLD to any 2-letter country code or those specified in the regex.  This isn't the most future-proof regex but it'll work for almost all emails used today.

vishalarora commented:
@wontoan done with this round of comments.

wontoan committed:
Login and Signup form validation + submission

-Create login.js + linked via application.js
-Update e-mail RegExp for cross-browser support.
-Submit form should work
-Change + update form class to “form”

wontoan committed:
Fix merge conflicts with style guide branch

wontoan committed:
Merge branch 'styleguide' into form-validation

Conflicts:
	app/assets/stylesheets/components/_form.scss
	app/assets/stylesheets/components/_inputs.scss

wontoan committed:
Remove margins from inputs.scss

wontoan committed:
Optimize forms for UX and responsiveness

-Changed password min to 6 characters.

wontoan committed:
8. Because vishal

wontoan committed:
Change 'remember' position in form

wontoan committed:
Update validation to include ' and - in names

wontoan committed:
Update validation to access DOM less

-Store reference in var = passValue
